Core/Core/Groups/GetGroups.swift

```diff
@@ -77,3 +77,32 @@ public struct GetDashboardGroups: CollectionUseCase {
 
     public let cacheKey: String? = "get-user-groups"
 }
+
+class GetGroupsInCategory: CollectionUseCase {
+    typealias Model = Group
+    let cacheKey: String?
+    let request: GetGroupsInCategoryRequest
+    let scope: Scope
+
+    init(_ groupCategoryID: String?) {
+        let groupCategoryID = groupCategoryID ?? ""
+        cacheKey = "group_categories/\(groupCategoryID)/groups"
+        request = GetGroupsInCategoryRequest(groupCategoryID: groupCategoryID)
+        scope = .where(
+            #keyPath(Group.groupCategoryID), equals: groupCategoryID,
+            orderBy: #keyPath(Group.name), naturally: true
+        )
+    }
+
+    func makeRequest(environment: AppEnvironment, completionHandler: @escaping ([APIGroup]?, URLResponse?, Error?) -> Void) {
+        // Skip making a request for empty groupCategoryID so this can be an empty list
```

Comment:
I moved the complexity of whether or not to make a request into this use case instead of having it live in the SwiftUI view. So the view just knows it has an empty list. `@ObservedObject` also cannot be Optional, so this was easier.